HubSpot V2 doc (#786)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/hubspot.mdx

---
title: "HubSpot"

> description: Step-by-step guide on sending your event data from RudderStack to HubSpot using the legacy API

Reviewer: ```suggestion
description: Step-by-step guide on sending your event data from RudderStack to HubSpot using the legacy API.
```

## Connection settings

To successfully configure HubSpot as a destination using the **legacy API**, you can choose any of the two authorization types explained below:

#### Private Apps authorization type

The **Private Apps** authorization type allows you to use HubSpot's APIs to access specific data from your HubSpot account.

<img src="../../assets/event-stream-destinations/hubspot-connection-settings-2.png" alt="HubSpot connection settings" />

- **Access Token**: Enter your HubSpot access token from the **Private Apps** section in the HubSpot dashboard.

<div class="infoBlock">
Refer to the <Link to="#faq">FAQ</Link> section to know where to get the HubSpot access token.
</div>

- **API Version**: Select **Legacy API** from the dropdown. To work with the new HubSpot API, refer to the <Link to="/destinations/streaming-destinations/hubspot-v2/">HubSpot V2</Link> guide.

#### API Key authorization type

The **API Key (deprecating soon)** authorization type allows you to create custom applications using HubSpot's APIs.

<div class="warningBlock">

> HubSpot API keys will be deprecated soon. Refer to <Link to="https://knowledge.hubspot.com/integrations/how-do-i-get-my-hubspot-api-key">HubSpot documentation</Link> for more information.

Reviewer: ```suggestion
This authorization type will be deprecated soon. Refer to <Link to="https://knowledge.hubspot.com/integrations/how-do-i-get-my-hubspot-api-key">HubSpot documentation</Link> for more information.
```

</div>

<img src="../../assets/event-stream-destinations/hubspot-connection-settings-1.png" alt="HubSpot connection settings" />

- **Hub ID**: Enter your HubSpot Hub ID.

> - **API Key**: Enter your HubSpot API key. This field is mandatory to send data to HubSpot via the RudderStack cloud mode.

Reviewer: ```suggestion
- **API Key**: Enter your HubSpot API key. This field is mandatory to send data to HubSpot via the RudderStack <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link>.
```

<div class="infoBlock">

> Refer to the <Link to="#faq">FAQ</Link> section to know where to get the Hub ID and API Key.

Reviewer: ```suggestion
Refer to the <Link to="#faq">FAQ</Link> section for more information on obtaining the Hub ID and API Key.
```

> A sample `identify` call for HubSpot CRM custom object is shown below:

Reviewer 2: Can we remove this custom objects section its planned as part of next release

Author: @utsabc this is the existing Hubspot doc (legacy API), just confirming if this will be released later in the next release?

Reviewer 2: > Can we remove this custom objects section its planned as part of next release

Reviewer 2: Yeah it was added long back but this functionality was never added. We are working on addressing it
